src/e2e/java/teammates/e2e/cases/InstructorSessionIndividualExtensionPageE2ETest.java

```diff
@@ -0,0 +1,207 @@
+package teammates.e2e.cases;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.DeadlineExtensionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.InstructorSessionIndividualExtensionPage;
+import teammates.e2e.util.TestProperties;
+
+/**
+ * SUT: {@link Const.WebPageURIs#INSTRUCTOR_SESSION_INDIVIDUAL_EXTENSION_PAGE}.
+ */
+public class InstructorSessionIndividualExtensionPageE2ETest extends BaseE2ETestCase {
+    private InstructorAttributes instructor;
+    private CourseAttributes course;
+    private FeedbackSessionAttributes feedbackSession;
+    private Collection<StudentAttributes> students;
+    private Collection<InstructorAttributes> instructors;
+    private String testEmail;
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorSessionIndividualExtensionPageE2ETest.json");
+        testEmail = TestProperties.TEST_EMAIL;
+        testData.students.get("alice.tmms@ISesIe.CS2104").setEmail(testEmail);
+        instructor = testData.instructors.get("ISesIe.instructor1");
+        course = testData.courses.get("course");
+        feedbackSession = testData.feedbackSessions.get("firstSession");
+        students = testData.students.values();
+        instructors = testData.instructors.values();
+
+        removeAndRestoreDataBundle(testData);
+    }
+
+    @Test
+    @Override
+    public void testAll() {
```

Comment:
Should we add:
* a test case where we extend some students and update some other students to a new timestamp X? So a mixed operation.
* a test case where users with and without extension are selected and we revoke extension: it will work on those with extension, and revocation won't be allowed if no extended users are selected

Reply:
Possible, but this is an e2e to testcase so we should choose which testcases to test carefully. From what I understand, e2e testcase is to verify the overall workflow and that components work well together rather than to test every possible branch. Need to choose some representative test cases rather than trying to cover all cases.

Reply:
To add on, from my understanding, the criteria for justifying a new e2e testcase should be whether the way the user interacts with the system is different. Covering the different edge cases might be better left to unit and integration tests.

Relevant discussion on this: https://github.com/TEAMMATES/teammates/pull/11613#discussion_r817712213

Reply:
I see. So the existing consensus is that it should cover the common use cases of the new feature. In that case, I believe this PR has done a great job! Not much to add then:)

Reply:
@moziliar I did mention the same thing to you before: https://github.com/TEAMMATES/teammates/pull/11567#issuecomment-1043686192.

In general, unit tests need to be designed with the mindset that E2E tests don't exist.